### PR TITLE
Add a strawman developer process doc

### PR DESCRIPTION
**Pull request by micahsaul**
Just trying to get something down in writing. Comments, changes, etc. more than welcome. Still needs style suggestions for Ruby and JS.

### CODE REVIEW

**Comment by rogeruiz** [comment]
I'm digging this. Should the language be stronger around merging PRs that contain your own commits? I think the `should` in the current readme implies that there _could_ be situations where merging your own PR _after_ it's been reviewed would be okay.

**Comment by micahsaul** [comment]
s/should/must ?

**Comment by rogeruiz** [comment]
Yeah I'm good with being explicit. Sets us up for consistency during handoffs. :)

File: DEVELOPERS.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+1. Does the code do what it claims to do?
+1. Are there unit tests? Are they passing?
+1. Do you understand what the code is doing? Remember, code is meant to be read by humans first.
+    1. Do the variable names make sense?
+    1. Has cleverness gotten in the way of clarity?
+    1. Is the code documented?
+1. Is the code idiomatic? This should be a softer requirement, but remember that for some languages (Python comes to mind) the notion of idiomatic code is a bit of a touchpoint.
+1. Was the code linted? If not, it should be!
+
+### Style Guides
```
Comment (juliasolorzano) [inline DEVELOPERS.md]: Maybe change to _Coding Style Guides_ or _Coding Best Practices_?

Comment (micahsaul) [inline DEVELOPERS.md]: I like best practices. Means we can extend it to things like: "Use Python3!"

**Comment by rogeruiz** [comment]
All this looks good. I want to add some JavaScript convention / styles once the dust settles in the Frontend Guild around JS style adoption.

**Pull request merged**